index.html

```diff
@@ -2175,6 +2231,130 @@ <h3>Verify Proof Sets and Chains</h3>
         </ol>
       </section>
 
+      <section class="normative">
+        <h3>Context Validation</h3>
+
+        <p>
+The purpose of the algorithm in this section is to ensure that a consuming
+application has explicitly approved of the types, and therefore the semantics,
+of input documents that it will process. Not checking JSON-LD context values
+against known good values can lead to security vulnerabilities, due to variance
+in the semantics that they convey. The context validation algorithm is one
+that applications validating a [=conforming secured document=] MUST run after
+running the algorithm in Section [[[#verify-proof]]] or Section
+[[[#verify-proof-sets-and-chains]]]. This algorithm takes a document ([=map=]
+|inputDocument|), a set of approved JSON-LD Contexts ([=map=]
+|approvedContexts|), and a boolean to recompact when unknown contexts are
+detected ([=boolean=] |recompact|) and returns a [=map=] that contains the
+following:
```

Comment:
I couldn't understand if the MUST meant that this particular algorithm MUST be run by any application that performs context validation (which I don't think should be the case), or if it meant that "this particular algorithm (if run), MUST be run after these other algorithms" (which makes sense to me).

```suggestion
in the semantics that they convey. This context validation algorithm is one that
applications MAY use to validate contexts in a [=conforming secured document=].
If used, it MUST be run after running the algorithm in Section [[[#verify-proof]]]
or Section [[[#verify-proof-sets-and-chains]]]. This algorithm takes a document
([=map=] |inputDocument|), a set of approved JSON-LD Contexts ([=map=]
|approvedContexts|), and a boolean to recompact when unknown contexts are
detected ([=boolean=] |recompact|) and returns a [=map=] that contains the
following:
```

Reply:
Building on @dlongley's suggestion
```suggestion
in the semantics that they convey. Applications MAY use this context validation
algorithm to validate contexts in a [=conforming secured document=]. If used,
it MUST be run <em>after</em> running the applicable algorithm in either
Section [[[#verify-proof]]] or Section [[[#verify-proof-sets-and-chains]]].
This algorithm takes inputs of a document ([=map=] |inputDocument|), a set of
approved JSON-LD Contexts ([=map=] |approvedContexts|), and a boolean to
recompact when unknown contexts are detected ([=boolean=] |recompact|), and
returns a [=map=] that contains the following:
```